hail/src/main/scala/is/hail/expr/types/physical/PNDArray.scala

```diff
@@ -1,224 +1,57 @@
-        )
-      },
-      index
-    )
-  }
-
-  def unlinearizeIndexRowMajor(index: Code[Long], shapeArray: Array[Code[Long]], region: Code[Region], mb: MethodBuilder): (Code[Unit], Array[Code[Long]]) = {
-    val nDim = shapeArray.length
-    val newIndices = (0 until nDim).map(_ => mb.newField[Long]).toArray
-    val elementsInProcessedDimensions = mb.newField[Long]
-    val workRemaining = mb.newField[Long]
-
-    val createShape = Code(
-      workRemaining := index,
-      elementsInProcessedDimensions := shapeArray.fold(const(1L))(_ * _),
-      Code.foreach(shapeArray.zip(newIndices)) { case (shapeElement, newIndex) =>
-        Code(
-          elementsInProcessedDimensions := elementsInProcessedDimensions / shapeElement,
-          newIndex := workRemaining / elementsInProcessedDimensions,
-          workRemaining := workRemaining % elementsInProcessedDimensions
-        )
-      }
-    )
-    (createShape, newIndices.map(_.load()))
-  }
-
-  def copyRowMajorToColumnMajor(rowMajorAddress: Code[Long], targetAddress: Code[Long], nRows: Code[Long], nCols: Code[Long], mb: MethodBuilder): Code[Unit] = {
-    val rowIndex = mb.newField[Long]
-    val colIndex = mb.newField[Long]
-    val rowMajorCoord = nCols * rowIndex + colIndex
-    val colMajorCoord = nRows * colIndex + rowIndex
-    val rowMajorFirstElementAddress = this.data.pType.firstElementOffset(rowMajorAddress, (nRows * nCols).toI)
-    val targetFirstElementAddress = this.data.pType.firstElementOffset(targetAddress, (nRows * nCols).toI)
-    val currentElement = Region.loadDouble(rowMajorFirstElementAddress + rowMajorCoord * 8L)
-
-    Code.forLoop(rowIndex := 0L, rowIndex < nRows, rowIndex := rowIndex + 1L,
-      Code.forLoop(colIndex := 0L, colIndex < nCols, colIndex := colIndex + 1L,
-        Region.storeDouble(targetFirstElementAddress + colMajorCoord * 8L, currentElement)
-      )
-    )
-  }
-
-  def copyColumnMajorToRowMajor(colMajorAddress: Code[Long], targetAddress: Code[Long], nRows: Code[Long], nCols: Code[Long], mb: MethodBuilder): Code[Unit] = {
-    val rowIndex = mb.newField[Long]
-    val colIndex = mb.newField[Long]
-    val rowMajorCoord = nCols * rowIndex + colIndex
-    val colMajorCoord = nRows * colIndex + rowIndex
-    val colMajorFirstElementAddress = this.data.pType.firstElementOffset(colMajorAddress, (nRows * nCols).toI)
-    val targetFirstElementAddress = this.data.pType.firstElementOffset(targetAddress, (nRows * nCols).toI)
-    val currentElement = Region.loadDouble(colMajorFirstElementAddress + colMajorCoord * 8L)
-
-    Code.forLoop(rowIndex := 0L, rowIndex < nRows, rowIndex := rowIndex + 1L,
-      Code.forLoop(colIndex := 0L, colIndex < nCols, colIndex := colIndex + 1L,
-        Region.storeDouble(targetFirstElementAddress + rowMajorCoord * 8L, currentElement)
-      )
-    )
-  }
+  def codeOrdering(mb: EmitMethodBuilder, other: PType): CodeOrdering = throw new UnsupportedOperationException
```

Comment:
not an override, so must be implemented. Kept John's solution to this, minus the override